### PR TITLE
use explicit types for commandline parsing

### PR DESCRIPTION
**Pull request by jmarolf**

### CODE REVIEW

File: src/FormatCommand.cs [hunk 1]
```diff
@@ -58,11 +62,84 @@ internal static RootCommand CreateCommandLineOptions()
...
+            {
+                // Use Defaults
+                return DiagnosticSeverity.Error;
+            }
+
+            var severityValue = result.Tokens.Single().Value;
+            if (!HasMatch(severityValue, out var severity))
+            {
+                // Assume that this means our arity is zero and the parser should re-parse accordingly
+                result.ErrorMessage = $"Cannot specify severity more than once for {result.Argument.Aliases.FirstOrDefault()}";
```
Comment (JoeRobich) [inline src/FormatCommand.cs:113]: Error message not applicable

File: src/FormatCommand.cs [hunk 2]
```diff
@@ -58,11 +62,84 @@ internal static RootCommand CreateCommandLineOptions()
...
+            {
+                result.ErrorMessage = "Cannot specify verbosity more than once.";
+                return null;
+            }
+
+            if (result.Tokens.Count == 0)
+            {
+                result.ErrorMessage = "Must specify a verbostiy level.";
+                return null;
+            }
```
Comment (JoeRobich) [inline src/FormatCommand.cs:77]: Are these checks necessary since we mark the argument arity as ExactlyOne?

File: src/FormatCommand.cs (hunk 1 above)
Comment (JoeRobich) [reply src/FormatCommand.cs:113]: Is this possible since we mark the argument with valid values using FromAmong?

File: src/FormatCommand.cs [hunk 3]
```diff
@@ -58,11 +62,84 @@ internal static RootCommand CreateCommandLineOptions()
...
+        private static DiagnosticSeverity? ParseDiagnosticSeverity(ArgumentResult result)
         {
-            return result.Tokens
-                .Where(token => token.Type == TokenType.Option)
-                .Any(token => aliases.Contains(token.Value));
+            if (result.Tokens.Count > 1)
+            {
+                result.ErrorMessage = $"Cannot specify severity more than once for {result.Argument.Name}";
+                return null;
+            }
```
Comment (JoeRobich) [inline src/FormatCommand.cs:101]: Is this necessary since we mark the argument arity as ZeroOrOne

File: src/FormatCommand.cs (hunk 1 above)
Comment (jmarolf) [reply src/FormatCommand.cs:84]: blocked by https://github.com/dotnet/command-line-api/pull/955

**Comment by jmarolf** [comment]
This would require us to write a completely custom help presenter. I think everything should stay stringly-typed for now

**Pull request closed without merging**